Fix errors reported by TypeScript 5.8

<!--
Comments in any language are welcome. Please write in your native language.
Se aceptan comentarios en cualquier idioma. Por favor, escribe en tu lengua materna.
欢迎使用任何语言发表评论。请用您的母语书写。
모든 언어로 된 댓글을 환영합니다. 모국어로 작성해 주세요.
どの言語も歓迎します。あなたの母国語で書いてください。
-->

- Markuplint version: `4.11.7`
- TypeScript version: `5.8.2`
- Node.js version: `v22.13.1`
- OS: Ubuntu 22.04.5 LTS

## Describe the bug

TypeScript reports errors in Markuplint. 

## Code Example or Playground URL


- `package.json`

  ```json
  {
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
      "markuplint": "4.11.7",
      "typescript": "5.8.2"
    }
  }
  ```

- `index.ts`

  ```typescript
  import { MLEngine } from "markuplint";

  console.log(MLEngine);
  ```

## Steps To Reproduce

1. `npm install`
2. `npx tsc --lib esnext,dom --module nodenext index.ts`

## The current behavior

```
node_modules/@markuplint/ml-core/lib/ml-dom/node/attr.d.ts:21:14 - error TS2416: Property 'ownerElement' in type 'MLAttr<T, O>' is not assignable to the same property in base type 'Attr'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

21     readonly ownerElement: MLElement<T, O>;
                ~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:26:9 - error TS2416: Property 'nextElementSibling' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

26     get nextElementSibling(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:35:9 - error TS2416: Property 'previousElementSibling' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

35     get previousElementSibling(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:39:5 - error TS2416: Property 'after' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

39     after(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:44:5 - error TS2416: Property 'before' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

44     before(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:55:5 - error TS2416: Property 'replaceWith' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

55     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/comment.d.ts:5:22 - error TS2420: Class 'MLComment<T, O>' incorrectly implements interface 'Comment'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
              Types of parameters 'nodes' and 'nodes' are incompatible.
                Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                  Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                    Type 'Node' is not assignable to type 'MLElement<any, any>'.
                      Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

5 export declare class MLComment<T extends RuleConfigValue, O extends PlainData = undefined> extends MLCharacterData<T, O, MLASTComment> implements Comment {
                       ~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-fragment.d.ts:5:22 - error TS2420: Class 'MLDocumentFragment<T, O>' incorrectly implements interface 'DocumentFragment'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
              Types of parameters 'nodes' and 'nodes' are incompatible.
                Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                  Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                    Type 'Node' is not assignable to type 'MLElement<any, any>'.
                      Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

5 export declare class MLDocumentFragment<T extends RuleConfigValue, O extends PlainData = undefined> extends MLParentNode<T, O, MLASTNode> implements DocumentFragment {
                       ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:25:5 - error TS2416: Property 'after' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

25     after(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:29:5 - error TS2416: Property 'before' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

29     before(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:37:5 - error TS2416: Property 'replaceWith' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

37     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:208:9 - error TS2416: Property 'doctype' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'DocumentType'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLElement<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLNode<any, any, MLASTNode>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
                Types of parameters 'nodes' and 'nodes' are incompatible.
                  Type 'string | Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
                    Type 'Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
                      Type 'Node' is missing the following properties from type 'MLNode<any, any, MLASTNode>': #private, MARKUPLINT_PREPROCESSOR_BLOCK, isFragment, rules, and 24 more.

208     get doctype(): MLDocumentType<T, O> | null;
            ~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1481:5 - error TS2416: Property 'getElementById' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(elementId: string) => MLElement<T, O>' is not assignable to type '(elementId: string) => HTMLElement'.
    Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'HTMLElement'.

1481     getElementById(elementId: string): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1488:5 - error TS2416: Property 'getElementsByClassName' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.
      Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1488     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1488:66 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1488     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                      ~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1502:5 - error TS2416: Property 'getElementsByTagName' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(qualifiedName: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '{ <K extends keyof HTMLElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<HTMLElementTagNameMap[K]>; <K extends keyof SVGElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<...>; <K extends keyof MathMLElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<...>; <K extends keyof HTMLElementDeprecatedTagNameMa...'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.
      Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1502     getElementsByTagName(qualifiedName: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1502:67 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1502     getElementsByTagName(qualifiedName: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                       ~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:403:9 - error TS2416: Property 'attributes' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type 'MLNamedNodeMap<T, O>' is not assignable to type 'NamedNodeMap'.
    The types of 'getNamedItem(...).ownerElement' are incompatible between these types.
      Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

403     get attributes(): MLNamedNodeMap<T, O>;
            ~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:403:9 - error TS2416: Property 'attributes' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type 'MLNamedNodeMap<T, O>' is not assignable to type 'NamedNodeMap'.
    The types of 'getNamedItem(...).ownerElement' are incompatible between these types.
      Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

403     get attributes(): MLNamedNodeMap<T, O>;
            ~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:600:9 - error TS2416: Property 'nextElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

600     get nextElementSibling(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:600:9 - error TS2416: Property 'nextElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

600     get nextElementSibling(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1465:9 - error TS2416: Property 'previousElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1465     get previousElementSibling(): MLElement<T, O> | null;
             ~~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1465:9 - error TS2416: Property 'previousElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1465     get previousElementSibling(): MLElement<T, O> | null;
             ~~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1596:5 - error TS2416: Property 'after' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLNode<any, any, MLASTNode>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
        Type 'Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
          Type 'Node' is missing the following properties from type 'MLNode<any, any, MLASTNode>': #private, MARKUPLINT_PREPROCESSOR_BLOCK, isFragment, rules, and 24 more.

1596     after(...nodes: (string | MLNode<any, any>)[]): void;
         ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1596:5 - error TS2416: Property 'after' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLNode<any, any, MLASTNode>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
        Type 'Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
          Type 'Node' is missing the following properties from type 'MLNode<any, any, MLASTNode>': #private, MARKUPLINT_PREPROCESSOR_BLOCK, isFragment, rules, and 24 more.

1596     after(...nodes: (string | MLNode<any, any>)[]): void;
         ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1628:5 - error TS2416: Property 'before' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

1628     before(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1628:5 - error TS2416: Property 'before' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

1628     before(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1708:5 - error TS2416: Property 'getAttributeNode' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Call signature return types 'MLAttr<T, O>' and 'Attr' are incompatible.
      The types of 'ownerElement' are incompatible between these types.
        Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1708     getAttributeNode(qualifiedName: string): MLAttr<T, O> | null;
         ~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1708:5 - error TS2416: Property 'getAttributeNode' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Call signature return types 'MLAttr<T, O>' and 'Attr' are incompatible.
      The types of 'ownerElement' are incompatible between these types.
        Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1708     getAttributeNode(qualifiedName: string): MLAttr<T, O> | null;
         ~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1750:5 - error TS2416: Property 'getElementsByClassName' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.
      Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1750     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1750:5 - error TS2416: Property 'getElementsByClassName' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.
      Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1750     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1750:66 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

1750     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                      ~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1831:5 - error TS2416: Property 'insertAdjacentElement' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(where: InsertPosition, element: MLElement<T, O>) => MLElement<T, O>' is not assignable to type '(where: InsertPosition, element: Element) => Element'.
    Types of parameters 'element' and 'element' are incompatible.
      Type 'Element' is missing the following properties from type 'MLElement<T, O>': #private, closeTag, elementType, isForeignElement, and 184 more.

1831     insertAdjacentElement(where: InsertPosition, element: MLElement<T, O>): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1831:5 - error TS2416: Property 'insertAdjacentElement' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(where: InsertPosition, element: MLElement<T, O>) => MLElement<T, O>' is not assignable to type '(where: InsertPosition, element: Element) => Element'.
    Types of parameters 'element' and 'element' are incompatible.
      Type 'Element' is missing the following properties from type 'MLElement<T, O>': #private, closeTag, elementType, isForeignElement, and 184 more.

1831     insertAdjacentElement(where: InsertPosition, element: MLElement<T, O>): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1905:5 - error TS2416: Property 'replaceWith' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

1905     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1905:5 - error TS2416: Property 'replaceWith' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
        Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
          Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

1905     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:4:5 - error TS2416: Property 'getNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Call signature return types 'MLAttr<T, O>' and 'Attr' are incompatible.
      The types of 'ownerElement' are incompatible between these types.
        Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

4     getNamedItem(qualifiedName: string): MLAttr<T, O> | null;
      ~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:12:5 - error TS2416: Property 'item' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(index: number) => MLAttr<T, O>' is not assignable to type '(index: number) => Attr'.
    Call signature return types 'MLAttr<T, O>' and 'Attr' are incompatible.
      The types of 'ownerElement' are incompatible between these types.
        Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

12     item(index: number): MLAttr<T, O> | null;
       ~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:19:5 - error TS2416: Property 'removeNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Call signature return types 'MLAttr<T, O>' and 'Attr' are incompatible.
      The types of 'ownerElement' are incompatible between these types.
        Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

19     removeNamedItem(qualifiedName: string): MLAttr<T, O>;
       ~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:26:5 - error TS2416: Property 'removeNamedItemNS' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(namespace: string, localName: string) => MLAttr<T, O>' is not assignable to type '(namespace: string, localName: string) => Attr'.
    Call signature return types 'MLAttr<T, O>' and 'Attr' are incompatible.
      The types of 'ownerElement' are incompatible between these types.
        Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

26     removeNamedItemNS(namespace: string | null, localName: string): MLAttr<T, O>;
       ~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:33:5 - error TS2416: Property 'setNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(attr: MLAttr<T, O>) => MLAttr<T, O>' is not assignable to type '(attr: Attr) => Attr'.
    Types of parameters 'attr' and 'attr' are incompatible.
      Type 'Attr' is missing the following properties from type 'MLAttr<T, O>': #private, endQuote, equal, isDuplicatable, and 37 more.

33     setNamedItem(attr: MLAttr<T, O>): MLAttr<T, O> | null;
       ~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:40:5 - error TS2416: Property 'setNamedItemNS' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(attr: MLAttr<T, O>) => MLAttr<T, O>' is not assignable to type '(attr: Attr) => Attr'.
    Types of parameters 'attr' and 'attr' are incompatible.
      Type 'Attr' is missing the following properties from type 'MLAttr<T, O>': #private, endQuote, equal, isDuplicatable, and 37 more.

40     setNamedItemNS(attr: MLAttr<T, O>): MLAttr<T, O> | null;
       ~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:135:9 - error TS2416: Property 'childNodes' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
    Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
      Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
        Types of property 'after' are incompatible.
          Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
            Types of parameters 'nodes' and 'nodes' are incompatible.
              Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                  Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

135     get childNodes(): NodeListOf<MLChildNode<T, O>>;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:135:34 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'Node'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
                Types of parameters 'nodes' and 'nodes' are incompatible.
                  Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                    Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                      Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

135     get childNodes(): NodeListOf<MLChildNode<T, O>>;
                                     ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:143:9 - error TS2416: Property 'firstChild' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
    Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
      Types of property 'after' are incompatible.
        Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
          Types of parameters 'nodes' and 'nodes' are incompatible.
            Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
              Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

143     get firstChild(): MLChildNode<T, O> | null;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:159:9 - error TS2416: Property 'lastChild' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
    Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
      Types of property 'after' are incompatible.
        Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
          Types of parameters 'nodes' and 'nodes' are incompatible.
            Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
              Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

159     get lastChild(): MLChildNode<T, O> | null;
            ~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:170:9 - error TS2416: Property 'nextSibling' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
    Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
      Types of property 'after' are incompatible.
        Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
          Types of parameters 'nodes' and 'nodes' are incompatible.
            Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
              Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

170     get nextSibling(): MLChildNode<T, O> | null;
            ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:211:9 - error TS2416: Property 'parentElement' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'HTMLElement'.

211     get parentElement(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:259:9 - error TS2416: Property 'parentNode' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLElement<T, O> | MLDocument<any, any> | MLDocumentFragment<any, any>' is not assignable to type 'ParentNode'.
    Type 'MLElement<T, O>' is not assignable to type 'ParentNode'.
      The types returned by 'children.namedItem(...)' are incompatible between these types.
        Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

259     get parentNode(): MLDocument<any, any> | MLDocumentFragment<any, any> | MLElement<T, O> | null;
            ~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:274:9 - error TS2416: Property 'previousSibling' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
    Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
      Types of property 'after' are incompatible.
        Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
          Types of parameters 'nodes' and 'nodes' are incompatible.
            Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
              Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

274     get previousSibling(): MLChildNode<T, O> | null;
            ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:373:41 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'Node'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
                Types of parameters 'nodes' and 'nodes' are incompatible.
                  Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                    Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                      Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

373     conditionalChildNodes(): NodeListOf<MLChildNode<T, O>>[];
                                            ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:380:5 - error TS2416: Property 'contains' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type '(other: MLNode<T, O, MLASTNode>) => boolean' is not assignable to type '(other: Node) => boolean'.
    Types of parameters 'other' and 'other' are incompatible.
      Type 'Node' is missing the following properties from type 'MLNode<T, O, MLASTNode>': #private, MARKUPLINT_PREPROCESSOR_BLOCK, isFragment, rules, and 24 more.

380     contains(other: MLNode<T, O> | null): boolean;
        ~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:401:37 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'Node'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
                Types of parameters 'nodes' and 'nodes' are incompatible.
                  Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                    Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                      Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

401     getPureChildNodes(): NodeListOf<MLChildNode<T, O>>;
                                        ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:409:5 - error TS2416: Property 'getRootNode' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type '(options?: GetRootNodeOptions) => MLNode<T, O, MLASTNode>' is not assignable to type '(options?: GetRootNodeOptions) => Node'.
    Call signature return types 'MLNode<T, O, MLASTNode>' and 'Node' are incompatible.
      The types of 'childNodes' are incompatible between these types.
        Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
          Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
            Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
              Types of property 'after' are incompatible.
                Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
                  Types of parameters 'nodes' and 'nodes' are incompatible.
                    Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                      Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                        Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

409     getRootNode(options?: GetRootNodeOptions): MLNode<T, O>;
        ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:509:43 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'Node'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
                Types of parameters 'nodes' and 'nodes' are incompatible.
                  Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                    Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                      Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

509     resetChildren(childNodes?: NodeListOf<MLChildNode<T, O>>): void;
                                              ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20:9 - error TS2416: Property 'children' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollection'.
    The types returned by 'namedItem(...)' are incompatible between these types.
      Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

20     get children(): HTMLCollectionOf<MLElement<T, O>>;
           ~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20:38 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

20     get children(): HTMLCollectionOf<MLElement<T, O>>;
                                        ~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:25:9 - error TS2416: Property 'firstElementChild' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

25     get firstElementChild(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:30:9 - error TS2416: Property 'lastElementChild' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Property 'ariaRelevant' is missing in type 'MLElement<T, O>' but required in type 'Element'.

30     get lastElementChild(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2523:5
    2523     ariaRelevant: string | null;
             ~~~~~~~~~~~~
    'ariaRelevant' is declared here.

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:56:53 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Node'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLElement<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLNode<any, any, MLASTNode>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
              Types of parameters 'nodes' and 'nodes' are incompatible.
                Type 'string | Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
                  Type 'Node' is not assignable to type 'string | MLNode<any, any, MLASTNode>'.
                    Type 'Node' is missing the following properties from type 'MLNode<any, any, MLASTNode>': #private, MARKUPLINT_PREPROCESSOR_BLOCK, isFragment, rules, and 24 more.

56     querySelectorAll(selectors: string): NodeListOf<MLElement<T, O>>;
                                                       ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/text.d.ts:5:22 - error TS2420: Class 'MLText<T, O>' incorrectly implements interface 'Text'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
              Types of parameters 'nodes' and 'nodes' are incompatible.
                Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                  Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                    Type 'Node' is not assignable to type 'MLElement<any, any>'.
                      Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, isForeignElement, and 310 more.

5 export declare class MLText<T extends RuleConfigValue, O extends PlainData = undefined> extends MLCharacterData<T, O, MLASTText> implements Text {
                       ~~~~~~


Found 61 errors in 11 files.

Errors  Files
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/attr.d.ts:21
     5  node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:26
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/comment.d.ts:5
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/document-fragment.d.ts:5
     3  node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:25
     6  node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:208
    19  node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:403
     6  node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:4
    13  node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:135
     5  node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/text.d.ts:5
```
## The expected behavior

No errors.

## Related issues

- https://github.com/markuplint/markuplint/issues/1889
- https://github.com/markuplint/markuplint/issues/1535
- https://github.com/markuplint/markuplint/issues/1124
- https://github.com/markuplint/markuplint/issues/1061

--- a/package.json
+++ b/package.json
@@ -89,7 +89,7 @@
 		"typedoc": "0.27.9",
 		"typedoc-plugin-mdn-links": "4.0.15",
 		"typedoc-plugin-resolve-crossmodule-references": "0.3.3",
-		"typescript": "5.7.3"
+		"typescript": "5.8.2"
 	},
 	"resolutions": {
 		"lerna/**/string-width": "4",

--- a/packages/@markuplint/create-rule/package.json
+++ b/packages/@markuplint/create-rule/package.json
@@ -30,7 +30,7 @@
 		"glob": "11.0.1",
 		"prettier": "3.5.2",
 		"ts-node": "10.9.2",
-		"typescript": "5.7.3"
+		"typescript": "5.8.2"
 	},
 	"devDependencies": {
 		"@types/fs-extra": "11.0.4",

--- a/packages/@markuplint/ml-core/src/ml-dom/node/element.ts
+++ b/packages/@markuplint/ml-core/src/ml-dom/node/element.ts
@@ -491,6 +491,17 @@ export class MLElement<T extends RuleConfigValue, O extends PlainData = undefine
 	 * @unsupported
 	 * @implements DOM API: `Element`
 	 */
+	get ariaRelevant(): string | null {
+		throw new UnexpectedCallError('Not supported "ariaRelevant" property');
+	}
+
+	/**
+	 * **IT THROWS AN ERROR WHEN CALLING THIS.**
+	 *
+	 * @deprecated
+	 * @unsupported
+	 * @implements DOM API: `Element`
+	 */
 	get ariaRoleDescription(): string | null {
 		throw new UnexpectedCallError('Not supported "ariaRoleDescription" property');
 	}

--- a/packages/@markuplint/pretenders/package.json
+++ b/packages/@markuplint/pretenders/package.json
@@ -32,6 +32,6 @@
 		"@markuplint/parser-utils": "4.8.5",
 		"glob": "11.0.1",
 		"meow": "13.2.0",
-		"typescript": "5.7.3"
+		"typescript": "5.8.2"
 	}
 }

--- a/vscode/package.json
+++ b/vscode/package.json
@@ -126,7 +126,7 @@
 		"@vscode/test-electron": "2.4.1",
 		"@vscode/vsce": "3.2.2",
 		"esbuild": "0.25.0",
-		"typescript": "5.7.3"
+		"typescript": "5.8.2"
 	},
 	"dependencies": {
 		"@markuplint/i18n": "4.6.0",

